src/systemcmds/top/CMakeLists.txt

```diff
@@ -33,7 +33,7 @@
 px4_add_module(
 	MODULE systemcmds__top
 	MAIN top
-	PRIORITY "SCHED_PRIORITY_MAX"
+	PRIORITY "SCHED_PRIORITY_DEFAULT"
```

Comment:
This was set to max on purpose. I had several cases where this helped.

Reply:
I was going to follow up with you. At SCHED_PRIORITY_MAX top can interfere with driver scheduling enough to cause an occasional fifo overflow and reset. I'd like to see if we can find a line in between.

Reply:
How about an argument to optionally run at max priority, but by default at least lower than the sensor bus WQ threads?

Reply:
At the time top was bumped to SCHED_PRIORITY_MAX the IMU drivers and a few other things were running out of ISRs (hrt_call_every).

Reply:
Continued in https://github.com/PX4/Firmware/pull/13115.

Reply:
Out of curiosity: is a fifo overflow handled gracefully?